packages/block-library/src/social-link/editor.scss

```diff
@@ -1,9 +1,20 @@
-.wp-block-social-links .wp-social-link button {
-	color: currentColor;
-	padding: 6px;
+// The editor uses the button component, the frontend uses a link.
+// Therefore we unstyle the button component to make it more like the frontend.
+.wp-block-social-links .wp-social-link {
+	line-height: 0;
+
+	button {
+		font-size: inherit;
+		color: currentColor;
+		height: auto;
+		line-height: 0;
+
+		// This rule is duplicated from the style.scss and needs to be the same as there.
+		padding: 0.25em;
+	}
 }
 
 .wp-block-social-links.is-style-pill-shape .wp-social-link button {
-	padding-left: 16px;
-	padding-right: 16px;
+	padding-left: 0.6666em;
+	padding-right: 0.6666em;
```

Comment:
Now, this is a very specific number 😄

Reply:
Heh, yes. The math checks out for the ems.

When the font size is 24px, 0.6666666666666 (infinite 6es but there wasn't space to write them all out) comes to 16px, which is the padding the pill shape should have left and right.

Reply:
I think in this case you can do `calc((2/3) * 1em)`

Reply:
That's not bad! Let me try that, thank you.